### ISSUE TITLE
Key Differences with Riffusion?

### ISSUE THREAD

**Issue by IFICL** [happylittlecat2333/Auffusion#2]
Hi, 
Thanks a ton for your work and for sharing this project with everyone. It's super helpful for the community! 

When reading the paper, I have several questions about it. The major concern I have is the key difference between Riffuision work. Riffusion didn't release its paper while they explained how they work on [their website](https://web.archive.org/web/20221229092306/https://www.riffusion.com/about) (the link does not exist anymore, so I use wayback machine to recall it). As I can tell from Riffusion, it's also finetuned from Stable diffusion and trained with paired mel-spectrogram and text descriptions. You also reimplemented this and trained with the same dataset as described in the supp. 

From my understanding of your work, Auffusion did several things additionally:
- Apply global normalization to the mel-spectrogram instead of individual normalization
- Change the text encoder to CLAP+FlanT5 instead of CLIP for text condition
- Using HiFiGAN vocoder to convert mel-spectrogram instead of [Griffin-Lim](https://web.archive.org/web/20221229092324/https://ieeexplore.ieee.org/document/1164317) algorithm

**I assume there might be more differences I didn't notice, perhaps the author can help identify them**. I also wonder which parts bring such large improvements. I understand HiFiGAN vocoder can bring a large improvement. From the experiment, the choices of text encoder only make a small difference. **Is global normalization also very helpful or just the HiFiGAN vocoder do all the work?** 

Looking forward to your reply.

**Comment by happylittlecat2333** [comment]
Hi,
Thank you for your attention! Our work is inspired by Riffusion, and we aim to demonstrate that pretrained T2I models possess strong cross-modal alignment with generative capabilities. Additionally, their abilities can be transferred to TTA tasks, achieving performance comparable to other TTA models. This approach can save significant time and resources compared to training from scratch.

The key differences between our model and Riffusion include:

1. Riffusion generates only 5-second music clips and hasn't been tested in TTA tasks. For a fair comparison, we reimplemented Riffusion to produce 10-second clips by adjusting the sample rate (from 44.1k to 16k) and hop size. We also use the same dataset to train our models and Riffusion.

2. Riffusion uses individual normalization and quantizes the spectrogram (float) into an image (int), which introduces a non-reversible process (individual normalization) and precision loss (float to int). As a result, their spectrogram conversion process cannot be reversed using the neural vocoder HiFi-GAN. Therefore, they adopt the Griffin-Lim algorithm, which is not sensitive to the initial numerical value of the generated spectrogram, because Griffin-Lim uses an iterative reverse process. In contrast, we adopt global normalization and input the spectrogram (float) directly into the VAE encoder. Therefore, we can use HiFi-GAN to convert back from the VAE decoder output (float). The whole process is a lossless audio conversion, without any precision loss or non-reversible process.

Thus, our carefully designed feature space transformation pipeline can fully utilize the capacity of T2I models. We also provide insightful assessments of text-audio alignment in TTA tasks, examining the impact of different text encoder choices through innovative cross-attention map visualizations between text and the generated spectrogram.

I hope my answer has addressed your concerns.

**Comment by IFICL** [comment]
Thank you for your response! To make sure I understand you correctly, do you mean that HiFiGAN cannot be applied if using individual normalization? If HiFGAN can be applied on the individual normalization version, do you test its performance?

**Comment by happylittlecat2333** [comment]
Yes. If you use individual normalization on spectrogram when finetune Stable Diffusion, you can not denormalize back to raw spectrogram. Therefore you can not use hifigan to convert raw spectrogram back to audio, since hifigan use <raw audio, raw spectrogram> pair for training.

**Comment by IFICL** [comment]
Gotcha. Thank you for your explanation.
